extensions/amp-story/1.0/live-story-manager.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AMP_LIVE_LIST_CUSTOM_SLOT_ID,
+} from '../../amp-live-list/0.1/live-list-manager';
+import {AmpEvents} from '../../../src/amp-events';
+import {createElementWithAttributes} from '../../../src/dom';
+import {dict} from '../../../src/utils/object';
+import {userAssert} from '../../../src/log';
+
+export class LiveStoryManager {
+  /**
+   * @param {!./amp-story.AmpStory} ampStory
+   */
+  constructor(ampStory) {
+    this.ampStory_ = ampStory;
+    this.storyEl_ = ampStory.element;
+  }
+
+  /**
+   * Initializes an amp-live-list component with the story-specific
+   * configuration and appends it to the DOM.
+   *
+   * @param {string} liveListId
+   */
+  build(liveListId) {
+    const liveListEl = createElementWithAttributes(this.ampStory_.win.document,
+        'amp-live-list', dict({
+          'id': liveListId,
+          'data-poll-interval':
+            this.storyEl_.getAttribute('data-poll-interval') || 15000,
+          'sort': 'ascending',
+          'disable-scrolling': '',
+          'disable-pagination': '',
+          'auto-insert': '',
+        }));
+    liveListEl[AMP_LIVE_LIST_CUSTOM_SLOT_ID] = userAssert(this.storyEl_.id,
+        'Story must contain id to build an amp-live-list');
+
+    this.storyEl_.insertBefore(liveListEl, this.storyEl_.firstElementChild);
+
+    this.storyEl_.addEventListener(AmpEvents.DOM_UPDATE, ({target}) => {
+      this.updateStory_(target);
+    });
+  }
+
+  /**
+   * Updates the client amp-story with the changes from the server document.
+   *
+   * @param {!Element} updatedStoryEl
+   * @private
+   */
+  updateStory_(updatedStoryEl) {
+    const newPages =
+    ([].slice.call(updatedStoryEl.querySelectorAll('amp-story-page')))
+        .filter(page => page.classList.contains('amp-live-list-item-new'));
+
+    const currentPages =
+      this.storyEl_.querySelectorAll('amp-story-page:not([ad])');
+    let lastPage = currentPages[currentPages.length - 1];
+
+    newPages.forEach(newPage => {
+      this.storyEl_.insertBefore(newPage, lastPage.nextElementSibling);
```

Comment:
This seems like an assumption we might not want to make (that there will always be another element after the last page of the story).  Can't we just insert after the `lastPage` instead of before the element after the `lastPage`?

Reply:
That's the idea actually, if `lastPage` doesn't have a sibling, then it just appends it right after `lastPage`. 

My reasoning for doing this is that we want all `<amp-story-page>` elements to be together.

In the case that there is a `<amp-story-bookend>`, for example, at the end of the set of pages, we want to append the `lasPage` after the last `<amp-story-page>` but before the `<amp-story-bookend>`

Reply:
Could you please use `dom.insertAfterOrAtStart`? It's exactly what you wrote but the name of the method better communicates how your code works :)